### ISSUE TITLE
Add a "vendor" top level namespace

### ISSUE THREAD

**Issue by remicollet** [phpmyadmin/sql-parser#126]
From http://www.php-fig.org/psr/psr-4/

> The fully qualified class name MUST have a top-level namespace name, also known as a “vendor namespace”.

Ok, you can say "SqlParser" is the vendor... a way to circumvent PSR-4 (as the project SubNamespaceName is not mandatory.. one reason why I dislike this PSR). And SqlParser could be a bit too generic and may create conflict.

Proposal: use PhpMyAdmin\SqlParser

I understand this is only doable as part of a new major version, because of BC break.

**Comment by nijel** [comment]
This pretty much applies to all libraries we ship, on the other side doing that before phpMyAdmin 4.7.0 is release would save us some headaches with possible rename in future.

Anyway this really needs feedback from other @phpmyadmin/developers ...

**Comment by ibennetch** [comment]
I don't have enough of an opinion to state a decision one way or the other.

I don't see any problem with doing this; even for other programs that use the library it shouldn't be a problem to reference the PhpMyAdmin namespace. On the other hand, I don't see why one would have two SQL parsing libraries with the same "SqlParser" namespace in one project, so I imagine the possibility of collision is quite low.

I'll go along with whatever Michal and the others decide, but have no reason to object to such a change.

**Comment by remicollet** [comment]
> This pretty much applies to all libraries we ship

Indeed.

> Anyway this really needs feedback

Yes, I open this issue mostly to raise such discussion. No urgency.

**Comment by nijel** [comment]
Well there is sort of urgency - 4.7.0 beta will be out probably next week and such change should be done prior to it (or wait for 4.8.0, what will postpone it for year or so).

**Comment by bigfoot90** [comment]
I think you should follow semantic versioning, and don't introduce BC break until 5.0.

**Comment by nijel** [comment]
This is about phpMyAdmin versions, not the library versions. Of course this would need major version increase on the library side.

**Comment by bigfoot90** [comment]
You're right.

**Comment by nijel** [comment]
In the end I think this is sane thing to do and should be done now - before phpMyAdmin 4.7.0 first beta is released. I've started to adjust the libraries right now.

**Comment by remicollet** [comment]
Great thanks for the work on SqlParser/MoTranslator/ShapFile, this will make things much more clean.

**Comment by nijel** [comment]
Okay, I think we're done. The change was done in phpMyAdmin master, the QA_4_6 branch stayed on the old version, in case new version would be needed for whatever reasons, there is phpmyadmin/phpmyadmin@e4bf53690679332c23d0ff2b4865ad844d60d996 which applies on QA_4_6, but needs deeper testing before it can be pushed to our users.
